source/Nevermore/Advanced/IConnectionAndTransaction.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nevermore.Advanced
+{
+    public interface IConnectionAndTransaction : IDisposable
+    {
+        DbConnection Connection { get; }
+        DbTransaction Transaction { get; }
+        void CommitTransaction();
+        Task CommitTransactionAsync(CancellationToken cancellationToken);
+    }
```

Comment:
The benefit of this interface is it gives the consumer full control over what happens on commit or dispose e.g. they may want to do some logging or something.